docs/guides/deployment-patterns/docker/docker-deployment.md

```diff
@@ -0,0 +1,146 @@
+
+### Verify Prerequisites
+
+```bash
+# Check Docker installation
+docker --version
+docker-compose --version
+
+# Test Docker daemon
+docker run hello-world
+
+# Check available system resources
+docker system df
+```
+
+## Quick Start
+
+Deploy Thunder with Docker in under 5 minutes:
+
+### Option 1: Run with Docker
+
+Follow these steps to run Thunder using Docker.
+
+1. **Pull the Docker image**
+
+    ```bash
+    docker pull ghcr.io/asgardeo/thunder:latest
+    ```
+
+2. **Run the container**
+
+    ```bash
+    docker run --rm \
+      -p 8090:8090 \
+      ghcr.io/asgardeo/thunder:latest
+    ```
+
+    Optionally if you want to modify the server configurations, you can mount a custom `deployment.yaml` file. Create a `deployment.yaml` file in your working directory similar to the [deployment.yaml](https://github.com/asgardeo/thunder/blob/main/backend/cmd/server/repository/conf/deployment.yaml), and mount it as below:
+
+    ```bash
+    docker run --rm \
+      -p 8090:8090 \
+      -v $(pwd)/deployment.yaml:/opt/thunder/repository/conf/deployment.yaml \
+      ghcr.io/asgardeo/thunder:latest
+    ```
+
+    Optionally if you want to use custom configurations or certificates, you can mount them as follows:
+
+    ```bash
+    docker run --rm \
+      -p 8090:8090 \
+      -v $(pwd)/deployment.yaml:/opt/thunder/repository/conf/deployment.yaml \
+      -v $(pwd)/certs/server.cert:/opt/thunder/repository/resources/security/server.cert \
+      -v $(pwd)/certs/server.key:/opt/thunder/repository/resources/security/server.key \
+      ghcr.io/asgardeo/thunder:latest
+    ```
+
+3. **Initialize with default data (Required for first-time setup)**
+
+    To use the Developer Console and sample applications, you need to set up initial data including the default admin user. Run this command after starting the container:
+
+    ```bash
+    docker exec -it $(docker ps -q --filter ancestor=ghcr.io/asgardeo/thunder:latest) \
+      /opt/thunder/scripts/setup_initial_data.sh
+    ```
+
+    This creates:
+    - Default admin user (`admin` / `admin`)  
+    - Developer Console application
+    - Basic authentication and registration flows
+
+    > [!NOTE]
+    > Replace the container selection with your specific container name if running multiple Thunder containers.
+
+### Access Thunder
+
+- **Application**: https://localhost:8090
+- **Gate (Login/Register)**: https://localhost:8090/signin
+- **Develop (Admin Console)**: https://localhost:8090/develop
+
+## Database Setup
+
+### Option 1: Embedded SQLite (Default)
+
+Thunder uses SQLite by default for development - no additional setup needed.
+
+### Option 2: External PostgreSQL
+
+For production deployments with PostgreSQL:
+
+#### Step 1: Start and Initialize PostgreSQL
+
+1. Navigate to local-development directory
+
+```bash
+cd install/local-development
+```
+
+2. Start PostgreSQL Database in background
+
+```bash
+docker compose up -d 
```

Comment:
Is this docker compose of the Database server ? rather than that can't we add the docker image here ?